cogs/ahk/ahk.py

```diff
@@ -944,6 +947,165 @@ async def on_member_remove(self, member: disnake.Member):
+                    continue
+
+                valid_attachments.append([attachment.filename, decoded])
+            return valid_attachments
+
+        valid_attachments = await _parse_attachments(message.attachments)
+
+        if not valid_attachments:
+            return
+
+        embed = disnake.Embed()
+        embed.title = "Share your file(s) using our paste service (p.autohotkey.com)?"
+        embed.description = (
+            "Click Yes if you want to upload them automatically.\n"
+            "This allows others to read your code without downloading files."
+        )
+        embed.color = disnake.Color.blue()
+
+        row = disnake.ui.ActionRow()
+
+        row.add_button(
+            style=disnake.ButtonStyle.primary,
+            custom_id="pahk_upload",
+            label="Upload",
+            emoji="☁️",
+        )
+
+        reply = await message.reply(embed=embed, components=[row])
+
+        def iter_checker(
+            author: disnake.Member | disnake.User,
+            channel: disnake.GroupChannel,
+            expected_custom_id: str,
+        ):
+            def check(inter: disnake.MessageInteraction) -> bool:
+                if inter.author != author:
+                    return False
+                if inter.channel != channel:
+                    return False
+                return (
+                    hasattr(inter.component, "custom_id")
+                    and inter.component.custom_id == expected_custom_id
+                )
+
+            return check
+
+        async def upload_to_pahk(attachments: list[str, str]) -> list[str, str]:
+            url = "https://pahk.geekdude.io/"
+            headers = {"Content-Type": "application/x-www-form-urlencoded"}
+            links = []
+            async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=False)) as session:
+                for filename, attachment in attachments:
+                    try:
+                        async with session.post(
+                            url, data=f"code={attachment}", headers=headers, allow_redirects=False
+                        ) as r:
+                            if r.status == 302 and r.headers.get("Location"):
+                                links.append(
+                                    [
+                                        filename,
+                                        f"https://p.autohotkey.com{r.headers['Location'][1:]}",
+                                    ]
+                                )
+                    except (aiohttp.ClientError, asyncio.TimeoutError):
+                        pass
+            return links
+
+        try:
+            check = iter_checker(message.author, message.channel, "pahk_upload")
+            await self.bot.wait_for(
+                event="button_click",
+                check=check,
+                timeout=180.0,
+            )
+        except asyncio.TimeoutError:
+            await reply.delete()
+            return None
```

Comment:
default return value of functions/methods is `None` so `return None` is implicit - `return` works fine!